pulp_smash/tests/pulp3/pulpcore/api_v3/test_api_docs.py

```diff
@@ -0,0 +1,51 @@
+# coding=utf-8
+"""Test related to the api docs page."""
+import unittest
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import HTTPError
+
+from pulp_smash import api, config, selectors, utils
+from pulp_smash.tests.pulp3.constants import API_DOCS_PATH
+from pulp_smash.tests.pulp3.pulpcore.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+
+class ApiDocsTestCase(unittest.TestCase, utils.SmokeTest):
+    """Test whether API auto generated docs are available.
+
+    This test targets the following issues:
+
+    * `Pulp #3552 <https://pulp.plan.io/issues/3552>`_
+    * `Pulp Smash #893 <https://github.com/PulpQE/pulp-smash/issues/893>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        if selectors.bug_is_untestable(3552, cls.cfg.pulp_version):
+            raise unittest.SkipTest('https://pulp.plan.io/issues/3552')
+
+    def setUp(self):
+        """Create an API Client."""
+        self.client = api.Client(self.cfg)
+
+    def test_access_api_docs_bad_auth(self):
+        """Test whether the API docs are available with bad credentials."""
+        self.client.request_kwargs['auth'] = HTTPBasicAuth(
+            utils.uuid4(),
+            utils.uuid4()
+        )
+        with self.assertRaises(HTTPError):
+            self.client.get(API_DOCS_PATH)
+
+    def test_access_api_docs_no_auth(self):
+        """Test whether the API docs page is available without auth."""
```

Comment:
This test doesn't do what's stated. `self.client` has authentication credentials.

Reply:
A new client is created for every test call. 

self.client.request_kwargs['auth']  is not used  to create authentication for each of these clients. Except in the case that bad credentials are used.

Reply:
> A new client is created for every test call.

Correct.

> `self.client.request_kwargs['auth']` is not used to create authentication for each of these clients. Except in the case that bad credentials are used.

Really?

```python
>>> from pulp_smash import api, config
>>> client = api.Client(config.get_config())
>>> client.request_kwargs['auth']
('admin', 'admin')
```